dali/pipeline/operator/eager_operator.h

```diff
@@ -165,9 +166,16 @@ class DLL_PUBLIC EagerOperator {
                        ", instance name: \"", name_, "\", encountered:\n", what);
   }
 
-  static inline int GetDefaultNumThreads() {
-    int num_cores = std::thread::hardware_concurrency();
-    return num_cores < 6 ? num_cores : 6;
+  static inline std::shared_ptr<ThreadPool> GetSharedThreadPool() {
+    if (!shared_thread_pool) {
```

Comment:
Is there any risk of a race condition?

Reply:
Probably we should add a lock to `GetSharedThreadPool` and `UpdateThreadPool`.

Reply:
done